src/modules/pos/validator/store/actions/staking.js

```diff
@@ -74,44 +74,55 @@ export const stakeEdited = (data) => async (dispatch) =>
  */
 export const stakesSubmitted =
   (formProps, transactionJSON, privateKey, _, senderAccount, moduleCommandSchemas) =>
-    async (dispatch, getState) => {
-      const state = getState();
-      const activeWallet = selectActiveTokenAccount(state);
-      const [error, tx] = await to(
-        signTransaction({
-          transactionJSON,
-          wallet: activeWallet,
-          schema: moduleCommandSchemas[joinModuleAndCommand(transactionJSON)],
-          chainID: state.network.networks.LSK.chainID,
-          privateKey,
-          senderAccount,
-        })
-      );
-      if (error) {
-        dispatch({
-          type: txActionTypes.transactionSignError,
-          data: error,
-        });
-      } else {
-        dispatch({ type: actionTypes.stakesSubmitted });
-        dispatch({
-          type: txActionTypes.transactionCreatedSuccess,
-          data: tx,
-        });
-      }
-    };
+  async (dispatch, getState) => {
+    const state = getState();
+    const activeWallet = selectActiveTokenAccount(state);
+    const [error, tx] = await to(
+      signTransaction({
+        transactionJSON,
+        wallet: activeWallet,
+        schema: moduleCommandSchemas[joinModuleAndCommand(transactionJSON)],
+        chainID: state.network.networks.LSK.chainID,
+        privateKey,
+        senderAccount,
+      })
+    );
+    if (error) {
+      dispatch({
+        type: txActionTypes.transactionSignError,
+        data: error,
+      });
+    } else {
+      dispatch({ type: actionTypes.stakesSubmitted });
+      dispatch({
+        type: txActionTypes.transactionCreatedSuccess,
+        data: tx,
+      });
+    }
+  };
 
 /**
  * Fetches the list of votes of the host wallet.
  */
 export const stakesRetrieved = () => async (dispatch, getState) => {
   const { network, account } = getState();
-  const address = account.current?.metadata?.address;
+  const currentAddress = account.current?.metadata?.address;
   try {
-    const votes = await getVotes({ network, params: { address } });
+    const stakes = await getStakes({ network, params: { address: currentAddress } });
+    const validators = await getValidatorList({
+      addresses: stakes.data.map(({ address }) => address),
+    });
+    const mapValidatorToAddress = validators.data.reduce(
```

Comment:
here the variable naming should be a resultant of what happened. E.g the outcome is a mapping of validator address to commission? so the correct name would be `validatorCommission`? for readability we should name batter, please consider these in next development 🙏🏼

Reply:
ok i think this requires refactoring, so ill create a separate issue.